src/common/libflux/message.c

```diff
@@ -165,52 +170,82 @@ static int proto_get_u32 (uint8_t *data, int len, int index, uint32_t *val)
     *val = ntohl (x);
     return 0;
 }
-static void proto_init (uint8_t *data, int len, uint8_t flags)
-{
-    int n;
-    assert (len >= PROTO_SIZE);
-    memset (data, 0, len);
-    data[PROTO_OFF_MAGIC] = PROTO_MAGIC;
-    data[PROTO_OFF_VERSION] = PROTO_VERSION;
-    data[PROTO_OFF_FLAGS] = flags;
-    n = proto_set_u32 (data, len, PROTO_IND_USERID, FLUX_USERID_UNKNOWN);
-    assert (n == 0);
-    n = proto_set_u32 (data, len, PROTO_IND_ROLEMASK, FLUX_ROLE_NONE);
-    assert (n == 0);
-}
 /* End manual codec
  */
 
+static void route_id_destroy (void *data)
+{
+    if (data) {
+        struct route_id *r = data;
+        free (r->id);
+        free (r);
+    }
+}
+
+static struct route_id *route_id_create (const char *id, unsigned int id_len)
+{
+    struct route_id *r = NULL;
+    if (!(r = calloc (1, sizeof (*r))))
```

Comment:
Don't need to set r to NULL right before you set it to the result of `calloc()`.  Let's not make gratuitous initialization a thing.